File: codigo/p1.java
/**
  Programa 1
  
  Eleva ao quadrado os N elementos de um vetor e calcula a soma.
 */

import java.util.Random;

public class p1 {

  public static void main (String[] args) {
    
    int n = Integer.parseInt(args[0]);
    float[] v = new float[n];
    Random rng = new Random();
    for (int i = 0; i < n; i++) {
      v[i] = rng.nextFloat();
    }
    Vetor a = new Vetor(v);
    System.out.println(a.somaQuadrados());
    
    
  }
  
}

class Vetor {
  
  private float[] valores;
  
  public Vetor(float[] valores) {
    this.valores = valores;
  }
  
  public float somaQuadrados() {
    float s = 0;
    
    for (float e : this.valores) {
      s += (e * e);
    }
    
    return s;
  }
  
}

File: codigo/p1.py
# Programa 1

# Eleva ao quadrado os N elementos de um vetor e calcula a soma.

import sys
import random

def soma_quadrados(valores):
  return sum([x**2 for x in valores])


def main(argv):
	n = int(argv)
	vetor = [random.random() for _ in range(n)]
	print(soma_quadrados(vetor))


if __name__ == "__main__":
	if (len(sys.argv)) != 2:
		print("Erro")
		sys.exit()

	main(sys.argv[1])
